### ISSUE TITLE
Models to Estimating Semantic Network 

### ISSUE THREAD

**Issue by hyiltiz** [tqian86/MPBNP#5]
I am sure this is not the place to ask about how is your model on estimating semantic network, but I think it is better than an email. I believe it is still a work in progress. I am quite interested in the collection (set) of categories that are used to construct the semantic network. Could you refer to me your methods, or program of your model?

BTW, maybe `Julia` is way quicker than `Python` and suitable to do OpenCL.

**Comment by tqian86** [comment]
Hi hyiltiz, thanks for the question. I'll defer to Joe to answer any question regarding semantic networks. Regarding Python vs Julia, we are aiming for the common denominator in the research community. Python is much more widely used than Julia. Also, the Python side of code in MPBNP is heavily vectorized and implemented via Numpy, which hooks to BLAS. I don't think Julia is somehow magically faster than a BLAS library, right? ;-) The so-called speed up in Julia usually applies to the limited cases of arithmetic loops - which have been carefully avoided as much as possible in the development of MPBNP.

**Comment by hyiltiz** [comment]
Yeah, BLAS rocks. I am really interested in the semantic networks and thank
you!

On Fri, Nov 13, 2015, 08:22 Ting Qian notifications@github.com wrote:

> Hi hyiltiz, thanks for the question. I'll defer to Joe to answer any
> question regarding semantic networks. Regarding Python vs Julia, we are
> aiming for the common denominator in the research community. Python is much
> more widely used than Julia. Also, the Python side of code in MPBNP is
> heavily vectorized and implemented via Numpy, which hooks to BLAS. I don't
> think Julia is somehow magically faster than a BLAS library, right? ;-) The
> so-called speed up in Julia usually applies to the limited cases of
> arithmetic loops - which have been carefully avoided as much as possible in
> the development of MPBNP.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/AusterweilLab/MPBNP/issues/5#issuecomment-156430445.
> 
> ## 

Sent from Gmail Mobile.
